### ISSUE TITLE
Prevent Primitive Type in name

### ISSUE THREAD

**Issue by Acollie** [mgechev/revive#1760]
**Is your feature request related to a problem? Please describe.**
For junior developers there is a temptation to write `sumInt`, however in
statically typed languages this is not needed.

**Describe the solution you'd like**
A lint rule (proposed name: `primitive-in-name`) that warns when a variable
name contains a predeclared primitive type name as a distinct word, e.g.:
- Int
- Bool
- String
- Float32 / Float64
- Rune
- Byte

```go
a := 5
b := 10
sumInt := a + b // <---- Warn: avoid primitive type in name
```

Detection should split identifiers on camelCase boundaries and match whole
segments (not substrings), so `Interface`, `Sprintf`, etc. aren't flagged.

**Considerations**
### What should happen for pointer-encoded names (`pUser`, `userPtr`)?
- Out of scope for v1, pointer-prefix naming is a separate, older convention
  and would warrant its own rule/discussion rather than folding it in here.

I am happy to work on this feature :)

**Comment by alexandear** [comment]
@Acollie, thank you for the proposal. Feel free to open a PR with POC.

Please make sure to check your rule on the following repos (from
https://github.com/alexandear/revive-bench/blob/86d06e7e6deab85db7e71ae4b04ddbd1316256db/targets/repos-all.txt#L1)

```
github.com/beego/beego
github.com/caddyserver/caddy
github.com/containers/podman
github.com/gin-gonic/gin
github.com/go-delve/delve
github.com/gohugoio/hugo
github.com/golangci/golangci-lint
github.com/google/go-github
github.com/goreleaser/goreleaser
github.com/grpc/grpc-go
github.com/kubernetes/client-go
github.com/mgechev/revive
github.com/oapi-codegen/oapi-codegen
github.com/prometheus/prometheus
github.com/rclone/rclone
github.com/traefik/traefik
github.com/tw93/Mole
github.com/twpayne/chezmoi
github.com/uber-go/fx
github.com/uber-go/goleak
github.com/uber-go/zap
```

**Comment by Acollie** [comment]
Thanks @alexandear 🫡 will do!

**Comment by Acollie** [comment]
@alexandear I have created a PR https://github.com/mgechev/revive/pull/1761 for the above changes
